evap/results/tests/test_views.py

```diff
@@ -732,216 +731,199 @@ def test_private_evaluation(self):
-
-    def test_textanswer_visibility_for_manager(self):
-        page = self.app.get("/results/semester/1/evaluation/1?view=full", user=self.manager)
-        self.assertIn(".general_orig_published.", page)
-        self.assertNotIn(".general_orig_hidden.", page)
-        self.assertNotIn(".general_orig_published_changed.", page)
-        self.assertIn(".general_additional_orig_published.", page)
-        self.assertNotIn(".general_additional_orig_hidden.", page)
-        self.assertIn(".general_changed_published.", page)
-        self.assertIn(".contributor_orig_published.", page)
-        self.assertIn(".contributor_orig_private.", page)
-        self.assertIn(".responsible_contributor_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_orig_hidden.", page)
-        self.assertNotIn(".responsible_contributor_orig_published_changed.", page)
-        self.assertIn(".responsible_contributor_changed_published.", page)
-        self.assertIn(".responsible_contributor_orig_private.", page)
-        self.assertNotIn(".responsible_contributor_orig_notreviewed.", page)
-        self.assertIn(".responsible_contributor_additional_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_hidden.", page)
+    def helper_test_general(self, user, view_general_results, textanswers_in):
 
+        textanswers_not_in = list(set(self.general_textanswers) - set(textanswers_in))
+        for con_view in self.contributor_views:
+            page = self.app.get(
+                f"/results/semester/1/evaluation/1?view_general_results={view_general_results.value}&view_contributor_results={con_view.value}",
+                user=user,
+            )
 
-class TestResultsTextanswerVisibility(WebTest):
-    fixtures = ["minimal_test_data_results"]
+            for answer in textanswers_in:
+                self.assertIn(answer, page)
+            for answer in textanswers_not_in:
+                self.assertNotIn(answer, page)
 
-    @classmethod
-    def setUpTestData(cls):
-        cache_results(Evaluation.objects.get(id=1))
+    def helper_test_contributor(self, user, view_contributor_results, textanswers_in):
+
+        textanswers_not_in = list(set(self.contributor_textanswers) - set(textanswers_in))
+        for gen_view in self.general_views:
+            page = self.app.get(
+                f"/results/semester/1/evaluation/1?view_contributor_results={view_contributor_results.value}&view_general_results={gen_view.value}",
+                user=user,
+            )
+
+            for answer in textanswers_in:
+                self.assertIn(answer, page)
+            for answer in textanswers_not_in:
+                self.assertNotIn(answer, page)
+
+    def test_manager(self):
+        with run_in_staff_mode(self):  # in staff mode, the manager can see everything
+            user = "manager@institution.example.com"
+
+            self.helper_test_general(
+                user,
```

Comment:
do you mean we should initialize all users that get tested (e.g. manager, student, contributor...) in the setUpTestData() method?
So that we reference them by self.manager, self.student...

Reply:
Yes, that was my intention. I just saw that they're part of the `minimal_test_data_results` fixture, when I wrote the comment I guess I thought that we create them in test setup anyways. Not that important, then, I think. Resolving.